src/core/basic_client.rs

```diff
@@ -1284,6 +1285,31 @@ impl BasicClient {
         }
     }
 
+    /// **[Capability Discovery Operation]** Check if attributes are supported.
+    ///
+    /// Checks if the given attributes are supported for the given type of operation.
+    ///
+    /// #Errors
+    ///
+    /// This operation never return a result but will terminate with Err(Success) or
+    /// Err(PsaErrorNotSupported) indicating whether the attributes are supported.
+    ///
+    /// See the operation-specific response codes returned by the service
+    /// [here](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/can_do_crypto.html#specific-response-status-codes).
+    pub fn can_do_crypto(&self, check_type: CheckType, attributes: Attributes) -> Result<()> {
+        let crypto_provider = self.can_provide_crypto()?;
+        let op = CanDoCrypto {
+            check_type,
+            attributes,
+        };
+        let _ = self.op_client.process_operation(
+            NativeOperation::CanDoCrypto(op),
+            crypto_provider,
+            &self.auth_data,
+        )?;
+        Ok(())
```

Comment:
The following should work as the return types are the same I think

```suggestion
        self.op_client.process_operation(
            NativeOperation::CanDoCrypto(op),
            crypto_provider,
            &self.auth_data,
        )
```

Reply:
The required return type for this function is `Result<()>` however `.process_operation` returns something of type `Result<NativeResult>` so it is probably better to keep it as it is. Unless it would be better to it your suggested method and convert it?

Reply:
You are totally correct, forgot about that 😅 Let's keep it this way then!